src/strongbox/config.clj

```diff
@@ -73,24 +109,27 @@
 (defn -load-settings
   "returns a map of user configuration settings that can be merged over `core/state`"
   [cli-opts file-opts etag-db]
-  (let [cfg (merge-config file-opts cli-opts)]
+  (let [cfg (merge-config file-opts cli-opts)
+        default-selected-addon-dir (->> cfg :addon-dir-list (map :addon-dir) first)]
     {:cfg cfg
      :cli-opts cli-opts
      :file-opts file-opts
      :etag-db etag-db
-     :selected-addon-dir (->> cfg :addon-dir-list (map :addon-dir) first)}))
+     :selected-addon-dir default-selected-addon-dir}))
 
 (defn-spec load-settings-file map?
   "reads application settings from the given file.
-  if file is missing or malformed, returns an empty map"
+  returns an empty map if file is missing or malformed."
   [cfg-file ::sp/file]
-  (when-not (fs/exists? cfg-file)
-    (warn "configuration file not found: " cfg-file))
-  (utils/load-json-file-safely cfg-file
-                               :no-file? {}
-                               :bad-data? {}
-                               :transform-map {:selected-catalogue keyword
-                                               :gui-theme keyword}))
+  (let [catalogue-source-list-transformer
```

Comment:
;; todo: incorporate into :transform-map if we have another case of transforming nested structures